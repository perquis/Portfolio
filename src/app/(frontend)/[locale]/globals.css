@tailwind base;
@tailwind components;
@tailwind utilities;
@import url(../../../styles/input.css);
@import url(../../../styles/themes.css);

:root {
  --tw-white: hsl(0, 0%, 100%);
  --tw-zinc-50: hsl(240, 5%, 96%);
  --tw-zinc-900: hsl(240, 4%, 16%);
  --tw-zinc-950: hsl(240, 6%, 10%);
  --tw-indigo-600: hsla(243, 75%, 59%, 0.75);
}

*:focus-visible {
  outline: 2px solid var(--tw-indigo-600) !important;
  outline-offset: 3px !important;
  box-sizing: border-box;
}

a {
  @apply rounded-sm;
}

::-moz-selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

::selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

html {
  scroll-behavior: smooth !important;
  scroll-padding-top: 64px;
}

.sandpack * {
  outline: none !important;
}

@layer base {
  :root {
    --background: 0 0% 100%;
    --foreground: 240 10% 3.9%;
    --card: 0 0% 100%;
    --card-foreground: 240 10% 3.9%;
    --popover: 0 0% 100%;
    --popover-foreground: 240 10% 3.9%;
    --primary: 240 5.9% 10%;
    --primary-foreground: 0 0% 98%;
    --secondary: 240 4.8% 95.9%;
    --secondary-foreground: 240 5.9% 10%;
    --muted: 240 4.8% 95.9%;
    --muted-foreground: 240 3.8% 46.1%;
    --accent: 240 4.8% 95.9%;
    --accent-foreground: 240 5.9% 10%;
    --destructive: 0 84.2% 60.2%;
    --destructive-foreground: 0 0% 98%;
    --border: 240 5.9% 90%;
    --input: 240 5.9% 90%;
    --ring: 240 10% 3.9%;
    --chart-1: 12 76% 61%;
    --chart-2: 173 58% 39%;
    --chart-3: 197 37% 24%;
    --chart-4: 43 74% 66%;
    --chart-5: 27 87% 67%;
    --radius: 0.5rem;
    --color-1: 0 100% 63%;
    --color-2: 270 100% 63%;
    --color-3: 210 100% 63%;
    --color-4: 195 100% 63%;
    --color-5: 90 100% 63%;
  }

  [data-theme="dark"] {
    --background: 240 10% 3.9%;
    --foreground: 0 0% 98%;
    --card: 240 10% 3.9%;
    --card-foreground: 0 0% 98%;
    --popover: 240 10% 3.9%;
    --popover-foreground: 0 0% 98%;
    --primary: 0 0% 98%;
    --primary-foreground: 240 5.9% 10%;
    --secondary: 240 3.7% 15.9%;
    --secondary-foreground: 0 0% 98%;
    --muted: 240 3.7% 15.9%;
    --muted-foreground: 240 5% 64.9%;
    --accent: 240 6% 10%;
    --accent-foreground: 0 0% 98%;
    --destructive: 0 62.8% 30.6%;
    --destructive-foreground: 0 0% 98%;
    --border: 240 3.7% 15.9%;
    --input: 240 3.7% 15.9%;
    --ring: 240 4.9% 83.9%;
    --chart-1: 220 70% 50%;
    --chart-2: 160 60% 45%;
    --chart-3: 30 80% 55%;
    --chart-4: 280 65% 60%;
    --chart-5: 340 75% 55%;
    --color-1: 0 100% 63%;
    --color-2: 270 100% 63%;
    --color-3: 210 100% 63%;
    --color-4: 195 100% 63%;
    --color-5: 90 100% 63%;
  }
}

@layer base {
  * {
    @apply border-border;
  }
  body {
    @apply bg-background text-foreground;
  }
}
