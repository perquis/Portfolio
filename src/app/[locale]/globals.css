@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --tw-white: hsl(0, 0%, 100%);
  --tw-zinc-50: hsl(240, 5%, 96%);
  --tw-zinc-900: hsl(240, 4%, 16%);
  --tw-zinc-950: hsl(240, 6%, 10%);
  --tw-indigo-600: hsla(243, 75%, 59%, 0.75);
}

*:focus-visible {
  outline: 2px solid var(--tw-indigo-600) !important;
  outline-offset: 3px !important;
  box-sizing: border-box;
}

::-moz-selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

::selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

@import url(../../styles/input.css);
@import url(../../styles/themes.css);

.sandpack * {
  outline: none !important;
}
