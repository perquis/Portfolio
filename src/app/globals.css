@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --tw-indigo-600: hsla(243, 75%, 59%, 0.75);
  --tw-rose-600: hsla(340, 82%, 55%, 0.75);
  --tw-white: hsl(0, 0%, 100%);
}

*:focus-visible {
  outline: 2px solid var(--tw-indigo-600) !important;
  outline-offset: 3px !important;
  box-sizing: border-box;
}

::-moz-selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

::selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

html[data-mode="dark"] *:focus-visible {
  outline: 2px solid var(--tw-rose-600) !important;
}

html[data-mode="dark"] ::-moz-selection {
  background: var(--tw-rose-600);
}

html[data-mode="dark"] ::selection {
  background: var(--tw-rose-600);
}
