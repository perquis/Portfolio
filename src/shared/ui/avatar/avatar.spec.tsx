import { expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { ComponentProps } from "react";

import type { Rounded, Size } from "@/interfaces/variants";

import Avatar from "./avatar";

jest.mock("next-intl/navigation", () => ({
  createSharedPathnamesNavigation: () => ({
    Link: "a",
  }),
}));

jest.mock("../../ui", () => ({
  Ratio: ({ children, ...props }: ComponentProps<"div">) => <div {...props}>{children}</div>,
}));

interface AvatarTestInput {
  rounded: Rounded;
  size: Size;
}

describe("Avatar", () => {
  it.each([
    { rounded: "default", size: "tiny" },
    { rounded: "default", size: "small" },
    { rounded: "default", size: "medium" },
    { rounded: "default", size: "large" },
    { rounded: "full", size: "tiny" },
    { rounded: "full", size: "small" },
    { rounded: "full", size: "medium" },
    { rounded: "full", size: "large" },
  ] as AvatarTestInput[])("renders correctly with rounded $rounded and size $size", ({ rounded, size }) => {
    const { asFragment } = render(
      <Avatar src="http://example.com/static/avatar.png" alt="User Avatar" rounded={rounded} size={size} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
