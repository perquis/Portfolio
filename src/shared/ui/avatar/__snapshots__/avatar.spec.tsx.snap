// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly with rounded default and size large 1`] = `
<div
  alt="User Avatar"
  className="overflow-hidden rounded-2xl h-16 w-16"
  resolution="1:1"
  src="http://example.com/static/avatar.png"
/>
`;

exports[`Avatar renders correctly with rounded default and size medium 1`] = `
<div
  alt="User Avatar"
  className="overflow-hidden rounded-xl h-12 w-12"
  resolution="1:1"
  src="http://example.com/static/avatar.png"
/>
`;

exports[`Avatar renders correctly with rounded default and size small 1`] = `
<div
  alt="User Avatar"
  className="overflow-hidden rounded-lg h-8 w-8"
  resolution="1:1"
  src="http://example.com/static/avatar.png"
/>
`;

exports[`Avatar renders correctly with rounded default and size tiny 1`] = `
<div
  alt="User Avatar"
  className="overflow-hidden rounded-md h-6 w-6"
  resolution="1:1"
  src="http://example.com/static/avatar.png"
/>
`;

exports[`Avatar renders correctly with rounded full and size large 1`] = `
<div
  alt="User Avatar"
  className="overflow-hidden rounded-full h-16 w-16"
  resolution="1:1"
  src="http://example.com/static/avatar.png"
/>
`;

exports[`Avatar renders correctly with rounded full and size medium 1`] = `
<div
  alt="User Avatar"
  className="overflow-hidden rounded-full h-12 w-12"
  resolution="1:1"
  src="http://example.com/static/avatar.png"
/>
`;

exports[`Avatar renders correctly with rounded full and size small 1`] = `
<div
  alt="User Avatar"
  className="overflow-hidden rounded-full h-8 w-8"
  resolution="1:1"
  src="http://example.com/static/avatar.png"
/>
`;

exports[`Avatar renders correctly with rounded full and size tiny 1`] = `
<div
  alt="User Avatar"
  className="overflow-hidden rounded-full h-6 w-6"
  resolution="1:1"
  src="http://example.com/static/avatar.png"
/>
`;
